api/main.py

```
@@ -119,98 +120,110 @@ async def chat():
-                        elif output.get("error") and "Error" in str(output.get("error")):
-                            status = "error"
-                        elif output.get("message") and "Error" in str(output.get("message")):
-                            status = "error"
+                    # NOTE: "output" in tool_end from langchain might be the raw return value
+                    output = tool_data.get("output")
+                    tool_name = tool_data.get("tool")
                     
-                    # Extract server name from tool mapping
-                    server_name = tool_name_to_server.get(action.tool, "Unknown")
+                    status = "approved"
+                    if isinstance(output, str) and ("Error" in output): status = "error"
```

Ruff (0.14.14)

[error] Multiple statements on one line (colon)

(E701)